### ISSUE TITLE
Function allowed as optional param value ? (Not so in VBx)

### ISSUE THREAD

**Issue by Kr00l** [WaynePhillipsEA/twinbasic#214]
**Describe the bug**
This is a scenario which is possible in tB but not in VBx.
tB allows it to define a function as the value of an optional parameter.
The question is does tB allow it by intent or by "accident" ?

**To Reproduce**
```
Module MainModule

    Public Sub Main()
        
        Call SubWithOptionalParam
        
    End Sub

Private Sub SubWithOptionalParam(Optional ByVal Value As Single = VBA.Rnd())
Debug.Print Value
End Sub
```

**Expected behavior**
If this has no intention and is by accident then it should throw an error like in VBx. see below screenshot.

**Screenshots**
VB6 throws error
![image](https://user-images.githubusercontent.com/33756666/121680271-a1f6d200-cab9-11eb-843a-f1f9dab8127c.png)

tB just proceeds
![image](https://user-images.githubusercontent.com/33756666/121680773-3bbe7f00-caba-11eb-8a79-77f57469d4ad.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version [0.10.1389]

**Additional context**
This feature enables some possibilities but also brings in some danger in regard to source code stability. It is therefore not ensured anymore that an optional param value is "deterministic". I also think this brings some issues when exposed to COM to the outside world. (ax-dll etc.)

**Comment by WaynePhillipsEA** [comment]
Indeed, this is currently allowed, and not currently restricted to constant expressions.  VB does actually allow a limited set of calls, such as the conversion routines e.g. Optional Value As Long = CLng("123")

**Comment by Kr00l** [comment]
If it's intended then my vote in this discussion is to keep it.
Like always there is a risk if mis-used. But otherwise the enhancement is useful.
What happens when using a custom function as default value and expose to a public method which is used then in another host such as VBA. Any harm there?
